src/app/components/application/application.component.html

```diff
@@ -1,16 +1,15 @@
 <div class="col-12 col-md-8">
   <router-outlet></router-outlet>
   <div class="text-end my-5">
-    <span *ngIf="showBackLink"><a href="javascript:void(0);"
-                                  [ngClass]="{disabled: isSaving}"
-                                  (click)="saveAndBack()"><i class="fa fa-long-arrow-left"></i> Previous Step</a></span>
+    <span *ngIf="showBackLink"><a href="javascript:void(0);" [ngClass]="{disabled: isSaving}" (click)="saveAndBack()"><i
+          class="fa fa-long-arrow-left"></i> Previous Step</a></span>
     <span *ngIf="showSaveAndQuit && showBackLink"> | </span>
     <a *ngIf="showSaveAndQuit" (click)="saveAndQuit()" [ngClass]="{disabled: isSaving}" href="javascript:void(0);">Save
       and Quit</a>
     <i class="fa fa-spinner fa-pulse fa-fw" *ngIf="isSaving"></i>
-    <span *ngIf="showNextLink"> | <a href="javascript:void(0);"
-                                     [ngClass]="{disabled: isSaving || disableNextLink}"
-                                     (click)="saveAndNext()">Save & Next <i
-      class="fa fa-long-arrow-right"></i></a></span>
+    <span *ngIf="showNextLink"> | <a href="javascript:void(0);" class="save-next-link"
+        [ngClass]="{disabled: isSaving || disableNextLink}" (click)="isSaving || disableNextLink || saveAndNext()">
```

Comment:
after some discussion I understand why. :).

Reply:
I'm still curious why this isn't in the `saveAndNext()` function instead of in the template. Doesn't the function already check `isSaving`?

Reply:
i think the logic for disable was already there but on hover, we dont want to enable that hyperlink in mobile version. So I kept it disabled in the action

Reply:
I'm specifically talking about `isSaving || disableNextLink || saveAndNext()`. Is there a reason you didn't do that check in the function instead of the template?

Reply:
because looking at the function, it seems to already be covered:
```typescript
  saveAndNext(): void {
    if (this.isSaving === false && this.disableNextLink === false) {
      this.router.navigate([this.routerLinks[this.routerIndex + 1].routerLink]);
      this.appDataService.saveApplication();
    }
  }
```

Reply:
Yes the function already does this but in mobile it will have the hyperlink enabled so I have to keep in action to avoid that on the front end.

Reply:
do they not effectively do the same thing?

Reply:
Yes they do the same thing but on the html action side it will disable the save&next for the mobile side otherwise it will be clickable

Reply:
ok, I think I get your reasoning now.